src/Box.java

```diff
@@ -1,60 +1,47 @@
-import java.util.ArrayList;
+import java.util.*;
 
 public class Box {
```

Comment:
Box only needs the set of possible numbers.  The other fields can just be derived.